specification/k3po.control/SPEC.md

```diff
@@ -0,0 +1,195 @@
+# K3po Control Protocol
+
+## Abstract
+
+This document defines the K3po control protocol used to communicate with a remote K3po server.
+
+## Requirements
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY",
+and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://tools.ietf.org/html/rfc2119).
+
+## The Protocol
+
+The protocol is made up of Commands and Events.  Commands are actions that can be triggered by an external entity.
+Events are triggered by observed actions or state changes and are sent back to the external entity.  The protocol 
+is formatted by
+
+TODO normalize
+
+```
+Command / Event Name\n
+headers
+empty line
+content
+
+```
+
+headers are key value pairs separated by a ":" and delimited by a new line "\n"
+
+If the "content-length" header is specified, then optional content of the specified length can be added.
+
+Headers that are not understood / processed by a command can be ignored.
```

Comment:
Message body is expected only for commands PREPARE, FINISHED and ERROR (see line 188 in the same file). In case you will add it to another type of command, it will be interpreted as a new command. For example:
**START

this is the body.**
will be considered as a START command followed by an incorrect command 'this' that will return an error.
The problem is that there is no way to signal that a command is finished (that it contains or not a body). So the decision is taken based on the command type (those 3 only) and on the presence of content-length header. For any other case, the body is considered an error. Should I specify this in the specification ?